pkg/database/token.go

```diff
@@ -336,23 +336,57 @@ func (db *Database) updateStatsCodeInvalid(t time.Time, authApp *AuthorizedApp)
 				SET codes_invalid = authorized_app_stats.codes_invalid + 1
 		`
 	if err := db.db.Exec(authAppSQL, t, authApp.ID).Error; err != nil {
-		db.logger.Errorw("failed to update authorized app stats", "error", err)
+		db.logger.Errorw("failed to update authorized app stats code invalid", "error", err)
 	}
 }
 
 // updateStatsCodeClaimed updates the statistics, increasing the number of codes
 // claimed.
-func (db *Database) updateStatsCodeClaimed(t time.Time, authApp *AuthorizedApp) {
-	t = timeutils.UTCMidnight(t)
+func (db *Database) updateStatsCodeClaimed(t time.Time, authApp *AuthorizedApp, vc *VerificationCode) {
```

Comment:
I think this function could be simplified by updating the two stats independently, potentially even updating CodeClaimDistribution in another function.

Similarly, this will be faster and less error-prone to do it in straight SQL:

```sql
WITH arr AS (
  -- this is the fun part. you could build this with case statements or use a 
  -- range function: https://www.postgresql.org/docs/9.3/rangetypes.html
)
INSERT INTO realm_stats(date, realm_id, code_claim_distribution
  VALUES ($1, $2, arr)
ON CONFLICT (date, realm_id) DO UPDATE
  SET code_claim_distribution = array_agg(UNNEST(arr) + UNNEST(realm_stats.code_claim_distribution))
```